src/legacy/core_plugins/kibana/migrations/is_doc.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { Doc } from './types';
+
+export function isDoc(doc: { [key: string]: unknown } | Doc): doc is Doc {
+  return (
+    typeof doc.id === 'string' &&
+    typeof doc.type === 'string' &&
+    doc.attributes !== undefined &&
+    typeof doc.attributes === 'object' &&
```

Comment:
When `typeof doc.attributes === 'object'` is `true`, `.attributes` cannot be `undefined` (line above), but can be `null`, I'm just curious if on line 26 you wanted to compare to `null` instead

```js
doc.attributes !== null
```